### PR TITLE
[3.5] bpo-30675: Fix multiprocessing code in regrtest

### PR DESCRIPTION
**Pull request by vstinner**
* Rewrite code to pass slaveargs from the master process to worker
  processes: reuse the same code of the Python master branch
* Move code to initialize tests in a new setup_tests() function,
  similar change was done in the master branch
* In a worker process, call setup_tests() with the namespace built
  from slaveargs to initialize correctly tests

Before this change, warm_caches() was not called in worker processes
because the setup was done before rebuilding the namespace from
slaveargs. As a consequence, the huntrleaks feature was unstable. For
example, test_zipfile reported randomly false positive on reference
leaks.

### CODE REVIEW

File: Lib/test/regrtest.py [hunk 1]
```diff
@@ -468,6 +467,45 @@ def run_test_in_subprocess(testname, ns):
...
+            pass
+        else:
+            for m in [msvcrt.CRT_WARN, msvcrt.CRT_ERROR, msvcrt.CRT_ASSERT]:
+                if ns.verbose and ns.verbose >= 2:
+                    msvcrt.CrtSetReportMode(m, msvcrt.CRTDBG_MODE_FILE)
+                    msvcrt.CrtSetReportFile(m, msvcrt.CRTDBG_FILE_STDERR)
+                else:
+                    msvcrt.CrtSetReportMode(m, 0)
+    if ns.wait:
+        input("Press any key to continue...")
```
Comment (serhiy-storchaka) [inline Lib/test/regrtest.py]: Does it make sense to do this on subprocesses?

File: Lib/test/regrtest.py [hunk 2]
```diff
@@ -509,58 +547,34 @@ def main(tests=None, **kwargs):
...
+        ns = types.SimpleNamespace(**ns_dict)
+
+        setup_tests(ns)
+
+        if ns.huntrleaks:
             unittest.BaseTestSuite._cleanup = False
+
         try:
-            result = runtest(*args, **kwargs)
+            result = runtest(testname, ns.verbose, ns.quiet,
```
Comment (serhiy-storchaka) [inline Lib/test/regrtest.py:558]: `use_resources` is not passed.

Comment (vstinner) [reply Lib/test/regrtest.py:558]: Oh, nice catch, fixed.

It's a bug in my change. I copied the runtest() call from above, but above, there is a first "support.use_resources = ns.use_resources" call.

File: Lib/test/regrtest.py (hunk 1 above)
Comment (vstinner) [reply Lib/test/regrtest.py]: I don't think so. It's not done in worker processes in master neither, so I moved this code out of setup_tests() again.

**Comment by vstinner** [comment]
I fixed the two bugs that you spotted @serhiy-storchaka, thanks for your review.

**Pull request merged**